Support for NxtPX4v2

Minderring opened the pull request:
The NxtPX4v2 is an open-source hardware designed and maintened by [HKUST UAV-Group](https://github.com/HKUST-Aerial-Robotics/Nxt-FC). And it is produced by [MicoAir Tech.](https://micoair.com/).

Waiting for your reply!
Best Regard!

peterbarker commented on Tools/AP_Bootloader/board_types.txt:
```diff
@@ -266,6 +266,7 @@ AP_HW_MicoAir405v2                   1150
 AP_HW_ORAQF405PRO                    1155
 AP_HW_CBU_StampH743                  1156
 AP_HW_FOXEERF405_V2                  1157
+AP_HW_NxtPX4v2                       1159
```
I suggest creating a PR to reserve an ID in case this PR takes a while to get in.

Minderring replied:
Thank you for your suggestion. I have cerated the PR: [#26746](https://github.com/ArduPilot/ardupilot/pull/26746)

andyp1per commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/defaults.parm:
```diff
@@ -0,0 +1,9 @@
+
+#Serial Port defaults
+SERIAL4_PROTOCOL 42
+SERIAL6_PROTOCOL 23
```
You should add protocol 16 for SERIAL5

Minderring replied:
Okay! fixed.

andyp1per reviewed (approved):
LGTM but I think you need to update the board id and maybe set the ESC telemetry protocol

peterbarker commented:
@Minderring needs a rebase

Minderring commented:
The board ID has been merged and rebased. @peterbarker @andyp1per @Hwurzburg 
If there is anything that needs to be modified, please let me know.

peterbarker commented:
> The board ID has been merged and rebased. @peterbarker @andyp1per @Hwurzburg If there is anything that needs to be modified, please let me know.

I've marked it as MergeOnCIPass

Hwurzburg commented:
I need to review..will do so ASAP

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/hwdef.dat:
```diff
@@ -0,0 +1,164 @@
+define HAL_GPIO_C_LED_PIN 92
+
+define HAL_GPIO_LED_ON 1
+define HAL_GPIO_LED_OFF 0
```
remove this line...board will not build with this line!

Minderring replied:
Okay. I have remove the line: define HAL_GPIO_LED_OFF 0

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/README.md:
```diff
@@ -0,0 +1,92 @@
+# NxtPX4v2 Flight Controller
+
+The NxtPX4v2 is an open-source hardware designed and maintened by [HKUST UAV-Group](https://github.com/HKUST-Aerial-Robotics/Nxt-FC). And it is produced by [MicoAir Tech.](http://micoair.com/).
```
if open source should submit schematics,etc. to the ArduPilot/Schematics repo

Minderring replied:
Got it. I have submitted the PCB files to the ArduPilot/Schematics repo. [#10](https://github.com/ArduPilot/Schematics/pull/10)
Please take a look.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/README.md:
```diff
@@ -0,0 +1,92 @@
+## UART Mapping
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART2 (DMA-enabled)
```
```suggestion
 - SERIAL1 -> UART2 (MAVLink2,DMA-enabled)
```

Minderring replied:
Fixed.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/README.md:
```diff
@@ -0,0 +1,92 @@
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART2 (DMA-enabled)
+ - SERIAL2 -> UART4 (DMA-enabled)
```
```suggestion
 - SERIAL2 -> UART4 (MAVLink2, DMA-enabled)
```

Minderring replied:
Fixed.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/README.md:
```diff
@@ -0,0 +1,92 @@
+
+## RC Input
+
+The default RC input is configured on the UART5_RX inverted from the SBUS pin. Other RC  protocols  should be applied at other UART port such as UART2, UART4 or UART8, and set the protocol to receive RC data: `SERIALn_PROTOCOL=23` and change SERIAL6 _Protocol to something other than '23'
```
```suggestion
The default RC input is configured on the UART5. The SBUS pin is inverted and connected to RX5. Non SBUS,  single wire serial inputs can be directly tied to RX5 if SBUS pin is left unconnected. RC could  be applied instead at a different UART port such as UART2, UART4 or UART8, and set the protocol to receive RC data: `SERIALn_PROTOCOL=23` and change SERIAL6 _Protocol to something other than '23'
```
this assumes that the SBUS inverter does not interfere with RX5 if unconnected. This is the normal configuration for most boards. If not, I can propose a different wording.

Minderring replied:
Fixed. Thanks!

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/NxtPX4v2/README.md:
```diff
@@ -0,0 +1,92 @@
+
+## VTX Support
+
+The SH1.0-6P connector supports a DJI O3 Air Unit connection. Pin 1 of the connector is 12v so be careful not to connect this to a peripheral requiring 5v.
```
```suggestion
The SH1.0-6P connector supports a DJI Air Unit / HD VTX connection. Protocol defaults to DisplayPort. Pin 1 of the connector is 12v so be careful not to connect this to a peripheral requiring 5v.
```

Minderring replied:
Fixed.

Hwurzburg reviewed (changes requested):
a few tweaks and one major HWDEF issue that will prevent it being built
also the back view image needs to have TX4/RX4 labels instead of just TX/RX on the SERIAL2 connector

Minderring commented:
> a few tweaks and one major HWDEF issue that will prevent it being built also the back view image needs to have TX4/RX4 labels instead of just TX/RX on the SERIAL2 connector

Thank you for your suggestion! I have fixed them all. Please take a look. @Hwurzburg

Hwurzburg reviewed (approved):
LGTM

Minderring commented:
@peterbarker Hi! I have rebased and please take a look.

Hwurzburg commented:
@Minderring again, please describe the changes in the the force pushes since reviews were done...otherwise we will have to wait until we can find time to do a complete re-review

Minderring commented:
> @Minderring again, please describe the changes in the the force pushes since reviews were done...otherwise we will have to wait until we can find time to do a complete re-review

Sorry! Just a minor change since the last approved review.
 I corrected some errors in the back_view image regarding the AUX label, then I rebased it, and that was all.

Hwurzburg commented:
lets us know when testing is completed and will mark for DevCall and merge

Minderring commented:
> lets us know when testing is completed and will mark for DevCall and merge

@Hwurzburg Thanks! It was thoroughly tested last month and continues to perform well.
